### DISCUSSION TITLE
Computing semivariogram on a curved mesh

### DISCUSSION THREAD

**Discussion by avaruuser** [mmaelicke/scikit-gstat#154] [Q&A]
I'm working with data defined on a surface mesh (3D vertices and faces) and I want to identify gradients on this mesh. To do so, I thought about using a semivariogram, however, my surface mesh is strongly curved so I wanted to use geodesic distances instead of the euclidean ones that are usually used for computing the variogram. Is it somehow possible to use geodesic distances in the Variogram class instead of euclidean ones? I could fit the semivariogram with scipy curve_fit but I want to make use of the advantages that come with the fitting methods of the Variogram class. I would appreciate any help with this.

**Comment by mmaelicke** [comment]
Hi,

As a start, you can have a look at this issue: https://github.com/mmaelicke/scikit-gstat/issues/108

You find a hacky implementation of the haversine formula in there. There are also a lot of information from @MuellerSeb in there, how this task can be solved with GSTools. 

A possible workaround, when working with geographic coordinates is to transform them first into a cartesian coordinate system (ie. using PyProj). Depending on your data, this can be pretty accurate. You want to choose a CRS that fits the region of interest well. If you are working with global coverage, you can use something like EPSG:3857, which has global coverage. Not so sure about the precision right now, but for estimating a variogram, it should be fine. 
Feel free to ask again if I got you somehow wrong, or the referenced issue is not helpful for you
Best,

Mirko

**Reply by avaruuser** [reply]
Hi Mirko, thanks for your help. I am not working with geographic coordinates, I work on surfaces that look more like an asteroid. I guess, I'll simply use scipy's curve_fit TRF fitting procedure and specify the boundaries as suggested [here](https://scikit-gstat.readthedocs.io/en/latest/technical/fitting.html).
